### PR TITLE
Use `mismatch` in `getNFTokenMintData`

### PR DESCRIPTION
**Pull request by shawnxie999**
`finalIDs` is an array that contains one extra NFTokenID than `prevIDs`. Using `set_difference` to find the difference is not efficient. We can use `mismatch` to find out the additional NFTokenID.

### CODE REVIEW

File: src/etl/NFTHelpers.cpp [hunk 1]
```diff
@@ -95,21 +95,23 @@ getNFTokenMintData(ripple::TxMeta const& txMeta, ripple::STTx const& sttx)
...
-    throw std::runtime_error(msg.str());
+    // Find the first NFT ID that doesn't match.  We're looking for an
+    // added NFT, so the one we want will be the mismatch in finalIDs.
+    auto const diff = std::mismatch(finalIDs.begin(), finalIDs.end(), prevIDs.begin(), prevIDs.end());
+
+    // There should always be a difference so the returned finalIDs
+    // iterator should never be end().  But better safe than sorry.
+    if (finalIDs.size() != prevIDs.size() + 1 || diff.first == finalIDs.end() || !owner)
+    {
+        std::stringstream msg;
```
Comment (godexsoft) [inline src/etl/NFTHelpers.cpp]: Let's rewrite this with `fmt::format` please 🥇
You can search the project for examples of `fmt::format` use.

**Review by godexsoft — DISMISSED** [review]
LGTM but please rewrite msg with `fmt::format` 👍

**Review by godexsoft — DISMISSED** [review]
Nice one 👍

**Review by godexsoft — APPROVED** [review]

**Comment by shawnxie999** [comment]
@godexsoft Thank you for the review. I think this PR would be good to merge as soon as you feel it's okay.

**Pull request merged**